langchain4j-azure-cosmos-no-sql/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/no/sql/AzureCosmosDBNoSqlEmbeddingStore.java

```diff
@@ -0,0 +1,215 @@
+    public List<String> addAll(List<Embedding> embeddings) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(Collectors.toList());
+        addAllInternal(ids, embeddings, null);
+        return ids;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(Collectors.toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+
+        String embeddingKey = this.cosmosVectorEmbeddingPolicy.getVectorEmbeddings().get(0).getPath().substring(1);
+        String referenceEmbeddingString = Arrays.stream(referenceEmbedding.vectorAsList().toArray())
+                .map(Object::toString)
+                .collect(Collectors.joining(","));
+
+        String query = String.format("SELECT TOP %d *, VectorDistance(c.%s,[%s]) AS SimilarityScore FROM c ORDER By " +
+                "VectorDistance(c.%s,[%s])", maxResults, embeddingKey, referenceEmbeddingString, embeddingKey, referenceEmbeddingString);
+
+        CosmosPagedIterable<AzureCosmosDBNoSqlMatchedDocument> results = this.container.queryItems(query,
+                new CosmosQueryRequestOptions(), AzureCosmosDBNoSqlMatchedDocument.class);
+
+        return results.stream()
+                .map( item -> toEmbeddingMatch(null))
+                .collect(Collectors.toList());
+    }
+
+    private void addInternal(String id, Embedding embedding, TextSegment embedded) {
+        addAllInternal(singletonList(id), singletonList(embedding), embedded == null ? null : singletonList(embedded));
+    }
+
+    private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        if (isNullOrEmpty(ids) || isNullOrEmpty(embeddings)) {
+            log.info("do not add empty embeddings to Azure CosmosDB  Mongo vCore");
+            return;
+        }
+
+        ensureTrue(ids.size() == embeddings.size(), "ids size is not equal to embeddings size");
+        ensureTrue(embedded == null || embeddings.size() == embedded.size(), "embeddings size is not equal to embedded size");
+
+        PartitionKeyDefinition partitionKey = this.containerProperties.getPartitionKeyDefinition();
+        List<CosmosItemOperation> operations = new ArrayList<>(ids.size());
+        for (int i=0;i<ids.size();i++) {
+            operations.add(CosmosBulkOperations.getCreateItemOperation(
+                    toNoSqlDbDocument(ids.get(i), embeddings.get(i), embedded == null ? null : embedded.get(i)),
+                            new PartitionKey(ids.get(i))));
+        }
+
+        this.container.executeBulkOperations(operations);
+    }
+
+    public static class Builder {
```

Comment:
nit: It his required or can be just a `@Builder` above ctor?

Reply:
Added the `@Builder` above ctor